src/com/amazon/corretto/crypto/provider/EcParameters.java

```diff
@@ -0,0 +1,115 @@
+            // to get the key size and get the corresponding curve name from our own database. Future versions of java
+            // may restrict reflective access to private modules, so this functionality may break.
+            //
+            // https://github.com/corretto/corretto-11/blob/14c02261590b4dc01284888a7a51d39ff581ac8d/src/java.base/share/classes/sun/security/util/ECParameters.java#L121
+            // https://github.com/corretto/corretto-11/blob/14c02261590b4dc01284888a7a51d39ff581ac8d/src/java.base/share/classes/module-info.java#L301-L314
+            try {
+                Method getKeySize = paramSpec.getClass().getMethod("getKeySize");
+                Integer keySize = Integer.class.cast(getKeySize.invoke(paramSpec));
+                name = EcUtils.getNameByKeySize(keySize);
+            } catch (ReflectiveOperationException e) {
+                // pass, perhaps due to reflective access control restrictions. rely null check below to throw.
+            }
+        }
+
+        if (name == null) {
+            throw new InvalidParameterSpecException("Only ECParameterSpec and ECGenParameterSpec supported");
+        }
+
+        ecInfo = EcUtils.getSpecByName(name);
+        if (ecInfo == null) {
+            throw new InvalidParameterSpecException("Unknown curve: " + paramSpec);
+        }
+    }
+
+    protected void engineInit(byte[] params) throws IOException {
+        String name = null;
+        try {
+            name = EcUtils.getNameByEncoded(params);
+        } catch (RuntimeCryptoException e) {
+            // pass, handle via null check below
+        }
+        if (name == null) {
+            throw new IOException("Only named EcParameters supported");
+        }
+        ecInfo = EcUtils.getSpecByName(name);
+        if (ecInfo == null) {
+            throw new IOException("Unknown named curve: " + name);
+        }
+    }
+
+    protected void engineInit(byte[] params, String unused) throws IOException {
+        engineInit(params);
+    }
+
+    protected <T extends AlgorithmParameterSpec> T engineGetParameterSpec(Class<T> spec)
+            throws InvalidParameterSpecException {
+
+        if (spec.isAssignableFrom(ECParameterSpec.class)) {
+            return spec.cast(ecInfo.spec);
+        }
+
+        if (spec.isAssignableFrom(ECGenParameterSpec.class)) {
+            return spec.cast(new ECGenParameterSpec(ecInfo.name));
+        }
+
+        throw new InvalidParameterSpecException("Only ECParameterSpec and ECGenParameterSpec supported");
+    }
+
+    protected byte[] engineGetEncoded() throws IOException {
+        return ecInfo.encoded.clone();  // clone to avoid exposing static reference
+    }
+
+    protected byte[] engineGetEncoded(String encodingMethod) throws IOException {
+        return engineGetEncoded();
+    }
+
+    protected String engineToString() {
+        if (ecInfo == null) {
+            return "Not initialized";
```

Comment:
Throw a RuntimeException instead? This seems dodgy.

Reply:
this is intentional, and is intended to preserve behavioral compatibility with [SunJCE's implementation](https://github.com/corretto/corretto-11/blob/develop/src/java.base/share/classes/sun/security/util/ECParameters.java#L191).